src/hypothesis/core.py

```diff
@@ -142,16 +146,26 @@ def do_draw(self, data):
 
 
 def is_invalid_test(
-    name, original_argspec, generator_arguments, generator_kwargs
+    name, original_argspec, type_hints, generator_arguments, generator_kwargs
 ):
-    def invalid(message):
+    def invalid(message, exception=InvalidArgument):
         def wrapped_test(*arguments, **kwargs):
-            raise InvalidArgument(message)
+            raise exception(message)
         return wrapped_test
 
     if not (generator_arguments or generator_kwargs):
-        return invalid(
-            'given must be called with at least one argument')
+        if not type_hints:
+            return invalid(
+                '@given must be called with at least one argument, or used '
+                'to wrap a test with type hints.'
+            )
+        elif all(v.is_empty for v in type_hints.values()):
```

Comment:
> There's a specific and a more general problem with raising an error: in this case, it would break backwards compatibility to error on an unknown type.

I don't think that's correct. You only need to use the type to strategy conversion on arguments that are mandatory and not explicitly passed, so it only ever gets invoked in cases that are currently a TypeError anyway.

>  Hypothesis shouldn't make life harder in these case

Same answer. This is only a problem if you have those and are not passing a strategy explicitly.

> it should be natural to be explicit when the magic doesn't work

I agree. The problem is not being explicit, it's that the behaviour as you currently have it makes it highly non-obvious when the magic isn't working.